<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0" stdsetdef="1">
  <author></author>
  <comment></comment>
  <exportmacro></exportmacro>
  <class>MainForm</class>
  <widget class="QMainWindow" name="MainForm">
    <property name="geometry">
      <rect>
        <x>0</x>
        <y>0</y>
        <width>777</width>
        <height>565</height>
      </rect>
    </property>
    <property name="windowTitle">
      <string>GLMadness Level Editor</string>
    </property>
    <widget class="QWidget">
      <widget class="CIGLWidget" name="GLView">
        <property name="geometry">
          <rect>
            <x>222</x>
            <y>10</y>
            <width>543</width>
            <height>473</height>
          </rect>
        </property>        
      </widget>
      <widget class="QTabWidget" name="tabView">
        <property name="geometry">
          <rect>
            <x>11</x>
            <y>10</y>
            <width>200</width>
            <height>250</height>
          </rect>
        </property>
        <widget class="QWidget" name="tabLevel">
          <attribute name="title">
            <string>Global/Level</string>
          </attribute>
          <widget class="QWidget" name="layout17">
            <property name="geometry">
              <rect>
                <x>10</x>
                <y>10</y>
                <width>175</width>
                <height>205</height>
              </rect>
            </property>
            <layout class="QVBoxLayout">
              <property name="margin">
                <number>0</number>
              </property>
              <item>
                <layout class="QVBoxLayout">
                  <property name="margin">
                    <number>0</number>
                  </property>
                  <item>
                    <widget class="QLabel" name="textLabel1">
                      <property name="text">
                        <string>CubeMap</string>
                      </property>
                      <property name="wordWrap">
                        <bool>false</bool>
                      </property>
                    </widget>
                  </item>
                  <item>
                    <layout class="QHBoxLayout">
                      <property name="margin">
                        <number>0</number>
                      </property>
                      <item>
                        <widget class="QLineEdit" name="editCubeMap">                          
                          <property name="minimumSize">
                            <size>
                              <width>24</width>
                              <height>24</height>
                            </size>
                          </property>
                        </widget>
                      </item>
                      <item>
                        <widget class="QPushButton" name="buttonBrowseCubeMap">                          
                          <property name="minimumSize">
                            <size>
                              <width>24</width>
                              <height>24</height>
                            </size>
                          </property>
                          <property name="maximumSize">
                            <size>
                              <width>24</width>
                              <height>32767</height>
                            </size>
                          </property>
                          <property name="text">
                            <string>...</string>
                          </property>
                          <property name="toolTip" stdset="0">
                            <string>Browse Cube Map</string>
                          </property>
                        </widget>
                      </item>
                    </layout>
                  </item>
                </layout>
              </item>
              <item>
                <layout class="QHBoxLayout">
                  <property name="margin">
                    <number>0</number>
                  </property>
                  <item>
                    <widget class="QLabel" name="textGridSize">
                      <property name="text">
                        <string>Grid Size</string>
                      </property>
                      <property name="wordWrap">
                        <bool>false</bool>
                      </property>
                    </widget>
                  </item>
                  <item>
                    <widget class="QPushButton" name="buttonEditGridSize">
                      <property name="text">
                        <string>edit...</string>
                      </property>
                    </widget>
                  </item>
                </layout>
              </item>
              <item>
                <layout class="QHBoxLayout">
                  <property name="margin">
                    <number>0</number>
                  </property>
                  <item>
                    <widget class="QLabel" name="textLabel1_2">
                      <property name="text">
                        <string>Level Time (sec.)</string>
                      </property>
                      <property name="wordWrap">
                        <bool>false</bool>
                      </property>
                    </widget>
                  </item>
                  <item>
                    <widget class="QSpinBox" name="spinboxlevelTime">
                      <property name="maximum">
                        <number>1800</number>
                      </property>
                      <property name="minimum">
                        <number>10</number>
                      </property>
                    </widget>
                  </item>
                </layout>
              </item>
              <item>
                <spacer name="spacer3">
                  <property name="sizeHint">
                    <size>
                      <width>20</width>
                      <height>16</height>
                    </size>
                  </property>
                  <property name="sizeType">
                    <enum>QSizePolicy::Expanding</enum>
                  </property>
                  <property name="orientation">
                    <enum>Qt::Vertical</enum>
                  </property>
                </spacer>
              </item>
            </layout>
          </widget>
        </widget>
        <widget class="QWidget" name="tabGeometry">
          <attribute name="title">
            <string>Ground Geometry</string>
          </attribute>
          <widget class="QWidget" name="layout18">
            <property name="geometry">
              <rect>
                <x>10</x>
                <y>10</y>
                <width>175</width>
                <height>205</height>
              </rect>
            </property>
            <layout class="QVBoxLayout">
              <property name="margin">
                <number>0</number>
              </property>
              <item>
                <layout class="QVBoxLayout">
                  <property name="margin">
                    <number>0</number>
                  </property>
                  <item>
                    <widget class="QLabel" name="textLabel1_3">
                      <property name="text">
                        <string>Special Cell Modes</string>
                      </property>
                      <property name="wordWrap">
                        <bool>false</bool>
                      </property>
                    </widget>
                  </item>
                  <item>
                    <layout class="QHBoxLayout">
                      <property name="margin">
                        <number>0</number>
                      </property>
                      <item>
                        <widget class="QPushButton" name="buttonSetAsStart">
                          <property name="text">
                            <string>Start</string>
                          </property>
                        </widget>
                      </item>
                      <item>
                        <widget class="QPushButton" name="buttonSetAsFinish">
                          <property name="enabled">
                            <bool>true</bool>
                          </property>
                          <property name="text">
                            <string>Finish</string>
                          </property>
                        </widget>
                      </item>
                    </layout>
                  </item>
                  <item>
                    <layout class="QHBoxLayout">
                      <property name="margin">
                        <number>0</number>
                      </property>
                      <item>
                        <widget class="QPushButton" name="buttonSetAsIce">
                          <property name="text">
                            <string>Ice</string>
                          </property>
                        </widget>
                      </item>
                      <item>
                        <widget class="QPushButton" name="buttonSetAsGlue">
                          <property name="text">
                            <string>Glue</string>
                          </property>
                        </widget>
                      </item>
                    </layout>
                  </item>
                </layout>
              </item>
              <item>
                <layout class="QVBoxLayout">
                  <property name="margin">
                    <number>0</number>
                  </property>
                  <item>
                    <widget class="QLabel" name="textLabel1_4">
                      <property name="text">
                        <string>Smooth/Flatten</string>
                      </property>
                      <property name="wordWrap">
                        <bool>false</bool>
                      </property>
                    </widget>
                  </item>
                  <item>
                    <layout class="QHBoxLayout">
                      <property name="margin">
                        <number>0</number>
                      </property>
                      <item>
                        <widget class="QPushButton" name="buttonSmooth">
                          <property name="text">
                            <string>Classic s.</string>
                          </property>
                          <property name="checkable">
                            <bool>false</bool>
                          </property>
                        </widget>
                      </item>
                      <item>
                        <widget class="QPushButton" name="buttonVsmooth">
                          <property name="text">
                            <string>Vertical s.</string>
                          </property>
                        </widget>
                      </item>
                    </layout>
                  </item>
                  <item>
                    <layout class="QHBoxLayout">
                      <property name="margin">
                        <number>0</number>
                      </property>
                      <item>
                        <widget class="QPushButton" name="buttonHsmooth">
                          <property name="text">
                            <string>Horizontal s.</string>
                          </property>
                        </widget>
                      </item>
                      <item>
                        <widget class="QPushButton" name="buttonFlatten">
                          <property name="text">
                            <string>Flatten</string>
                          </property>
                        </widget>
                      </item>
                    </layout>
                  </item>
                </layout>
              </item>
              <item>
                <widget class="QPushButton" name="buttonDropBall">
                  <property name="text">
                    <string>Drop Ball</string>
                  </property>
                </widget>
              </item>
              <item>
                <spacer name="spacer2">
                  <property name="sizeHint">
                    <size>
                      <width>20</width>
                      <height>110</height>
                    </size>
                  </property>
                  <property name="sizeType">
                    <enum>QSizePolicy::Expanding</enum>
                  </property>
                  <property name="orientation">
                    <enum>Qt::Vertical</enum>
                  </property>
                </spacer>
              </item>
            </layout>
          </widget>
        </widget>
      </widget>
    </widget>
    <widget class="QToolBar" name="toolBar">
      <property name="windowTitle">
        <string>Tools</string>
      </property>
      <addaction name="fileNewAction"/>
      <addaction name="fileOpenAction"/>
      <addaction name="fileSaveAction"/>
      <addaction name="fileExitAction"/>
      <addaction name="viewResetTrack"/>
      <addaction name="editAction"/>
      <addaction name="trackballAction"/>
    </widget>
    <widget class="QMenuBar" name="MenuBar">
      <property name="minimumSize">
        <size>
          <width>40</width>
          <height>22</height>
        </size>
      </property>
      <widget class="QMenu" name="FileMenu">
        <property name="title">
          <string>&amp;File</string>
        </property>
        <addaction name="fileNewAction"/>
        <addaction name="fileOpenAction"/>
        <addaction name="fileSaveAction"/>
        <addaction name="separator"/>
        <addaction name="fileExitAction"/>
      </widget>
      <widget class="QMenu" name="helpMenu">
        <property name="title">
          <string>&amp;Help</string>
        </property>
        <addaction name="helpAboutAction"/>
      </widget>
      <widget class="QMenu" name="Menu">
        <property name="title">
          <string>Menu</string>
        </property>
        <addaction name="viewResetTrack"/>
      </widget>
      <addaction name="FileMenu"/>
      <addaction name="helpMenu"/>
      <addaction name="Menu"/>
    </widget>
    <action name="fileNewAction">
      <property name="icon">
        <iconset>images/filenew</iconset>
      </property>
      <property name="iconText">
        <string>New</string>
      </property>
      <property name="text">
        <string>&amp;New</string>
      </property>
      <property name="shortcut">
        <string>Ctrl+N</string>
      </property>
    </action>
    <action name="fileOpenAction">
      <property name="icon">
        <iconset>images/fileopen</iconset>
      </property>
      <property name="iconText">
        <string>Open</string>
      </property>
      <property name="text">
        <string>&amp;Open...</string>
      </property>
      <property name="shortcut">
        <string>Ctrl+O</string>
      </property>
    </action>
    <action name="fileSaveAction">
      <property name="icon">
        <iconset>images/filesave</iconset>
      </property>
      <property name="iconText">
        <string>Save</string>
      </property>
      <property name="text">
        <string>&amp;Save</string>
      </property>
      <property name="shortcut">
        <string>Ctrl+S</string>
      </property>
    </action>
    <action name="fileSaveAsAction">
      <property name="iconText">
        <string>Save As</string>
      </property>
      <property name="text">
        <string>Save &amp;As...</string>
      </property>
      <property name="shortcut">
        <string/>
      </property>
    </action>
    <action name="fileExitAction">
      <property name="icon">
        <iconset>images/exit.bmp</iconset>
      </property>
      <property name="iconText">
        <string>Exit</string>
      </property>
      <property name="text">
        <string>E&amp;xit</string>
      </property>
      <property name="shortcut">
        <string/>
      </property>
    </action>
    <action name="helpContentsAction">
      <property name="iconText">
        <string>Contents</string>
      </property>
      <property name="text">
        <string>&amp;Contents...</string>
      </property>
      <property name="shortcut">
        <string/>
      </property>
    </action>
    <action name="helpIndexAction">
      <property name="iconText">
        <string>Index</string>
      </property>
      <property name="text">
        <string>&amp;Index...</string>
      </property>
      <property name="shortcut">
        <string/>
      </property>
    </action>
    <action name="helpAboutAction">
      <property name="iconText">
        <string>About</string>
      </property>
      <property name="text">
        <string>&amp;About</string>
      </property>
      <property name="shortcut">
        <string/>
      </property>
    </action>
    <action name="viewResetTrack">
      <property name="icon">
        <iconset>images/home.png</iconset>
      </property>
      <property name="iconText">
        <string>Reset TrackBall</string>
      </property>
      <property name="text">
        <string>Reset TrackBall</string>
      </property>
    </action>
    <actiongroup name="ActionGroup">
      <action name="editAction">
        <property name="checkable">
          <bool>true</bool>
        </property>
        <property name="icon">
          <iconset>images/arrow.bmp</iconset>
        </property>
        <property name="iconText">
          <string>Edit mode</string>
        </property>
        <property name="text">
          <string>Edit mode</string>
        </property>
      </action>
      <action name="trackballAction">
        <property name="checkable">
          <bool>true</bool>
        </property>
        <property name="checked">
          <bool>true</bool>
        </property>
        <property name="icon">
          <iconset>images/track.bmp</iconset>
        </property>
        <property name="iconText">
          <string>Trackball Mode</string>
        </property>
        <property name="text">
          <string>Trackball Mode</string>
        </property>
      </action>
    </actiongroup>
  </widget>
  <layoutdefault spacing="6" margin="11"/>
  <customwidgets>
    <customwidget>
      <class>CIGLWidget</class>
      <extends>QWidget</extends>
      <header location="local">CIGLWidget.h</header>
      <sizehint>
        <width>-1</width>
        <height>-1</height>
      </sizehint>
      <container>0</container>      
      <pixmap>image0</pixmap>
    </customwidget>
  </customwidgets>  
  <includes>
    <include location="local">CIGLWidget.h</include>
  </includes>
  <connections>
    <connection>
      <sender>fileNewAction</sender>
      <signal>triggered()</signal>
      <receiver>MainForm</receiver>
      <slot>fileNew()</slot>
    </connection>
    <connection>
      <sender>fileOpenAction</sender>
      <signal>triggered()</signal>
      <receiver>MainForm</receiver>
      <slot>fileOpen()</slot>
    </connection>
    <connection>
      <sender>fileSaveAction</sender>
      <signal>triggered()</signal>
      <receiver>MainForm</receiver>
      <slot>fileSave()</slot>
    </connection>
    <connection>
      <sender>fileExitAction</sender>
      <signal>triggered()</signal>
      <receiver>MainForm</receiver>
      <slot>fileExit()</slot>
    </connection>
    <connection>
      <sender>helpAboutAction</sender>
      <signal>triggered()</signal>
      <receiver>MainForm</receiver>
      <slot>helpAbout()</slot>
    </connection>
    <connection>
      <sender>buttonEditGridSize</sender>
      <signal>released()</signal>
      <receiver>MainForm</receiver>
      <slot>editGridSize()</slot>
    </connection>
    <connection>
      <sender>editAction</sender>
      <signal>toggled(bool)</signal>
      <receiver>GLView</receiver>
      <slot>setEditing(bool)</slot>
    </connection>
    <connection>
      <sender>buttonBrowseCubeMap</sender>
      <signal>clicked()</signal>
      <receiver>MainForm</receiver>
      <slot>chooseCubeMap()</slot>
    </connection>
    <connection>
      <sender>editCubeMap</sender>
      <signal>returnPressed()</signal>
      <receiver>MainForm</receiver>
      <slot>setCubeMap()</slot>
    </connection>
    <connection>
      <sender>buttonSmooth</sender>
      <signal>released()</signal>
      <receiver>GLView</receiver>
      <slot>smooth()</slot>
    </connection>
    <connection>
      <sender>buttonFlatten</sender>
      <signal>released()</signal>
      <receiver>GLView</receiver>
      <slot>flatten()</slot>
    </connection>
    <connection>
      <sender>buttonDropBall</sender>
      <signal>released()</signal>
      <receiver>MainForm</receiver>
      <slot>setBallPos()</slot>
    </connection>
    <connection>
      <sender>buttonSmooth</sender>
      <signal>released()</signal>
      <receiver>MainForm</receiver>
      <slot>smooth()</slot>
    </connection>
    <connection>
      <sender>buttonFlatten</sender>
      <signal>released()</signal>
      <receiver>MainForm</receiver>
      <slot>flatten()</slot>
    </connection>
    <connection>
      <sender>buttonSetAsStart</sender>
      <signal>released()</signal>
      <receiver>MainForm</receiver>
      <slot>setAsStart()</slot>
    </connection>
    <connection>
      <sender>buttonSetAsFinish</sender>
      <signal>released()</signal>
      <receiver>MainForm</receiver>
      <slot>setAsFinish()</slot>
    </connection>
    <connection>
      <sender>buttonSetAsIce</sender>
      <signal>released()</signal>
      <receiver>MainForm</receiver>
      <slot>setAsIce()</slot>
    </connection>
    <connection>
      <sender>buttonSetAsGlue</sender>
      <signal>released()</signal>
      <receiver>MainForm</receiver>
      <slot>setAsGlue()</slot>
    </connection>
    <connection>
      <sender>buttonVsmooth</sender>
      <signal>released()</signal>
      <receiver>MainForm</receiver>
      <slot>vSmooth()</slot>
    </connection>
    <connection>
      <sender>buttonHsmooth</sender>
      <signal>released()</signal>
      <receiver>MainForm</receiver>
      <slot>hSmooth()</slot>
    </connection>
    <connection>
      <sender>viewResetTrack</sender>
      <signal>triggered()</signal>
      <receiver>MainForm</receiver>
      <slot>resetTrackBall()</slot>
    </connection>
  </connections>
</ui>
